inc/form.class.php

```diff
@@ -646,8 +646,12 @@ public function showFormList($rootCategory = 0, $keywords = '', $helpdeskHome =
             'contains' => $keywords
       ]);
       if ($selectedCategories != '') {
-         $query_faqs = "SELECT * FROM ($query_faqs)  AS `faqs`
+         $query_faqs = "SELECT `faqs`.* FROM ($query_faqs)  AS `faqs`
          WHERE `faqs`.`knowbaseitemcategories_id` IN (SELECT `knowbaseitemcategories_id` FROM `$table_cat` WHERE `id` IN ($selectedCategories) AND `knowbaseitemcategories_id` <> '0')";
+      } else {
+         $query_faqs = "SELECT `faqs`.* FROM ($query_faqs)  AS `faqs`
+         LEFT JOIN `$table_cat` ON (`faqs`.`knowbaseitemcategories_id` = `$table_cat`.`knowbaseitemcategories_id`)
```

Comment:
I checked the table ```glpi_knowbaseitems``` : knowbaseitemcategories_id cannot be null because the table does not allows this

```
CREATE TABLE `glpi_knowbaseitems` (
	`id` INT(11) NOT NULL AUTO_INCREMENT,
	`knowbaseitemcategories_id` INT(11) NOT NULL DEFAULT '0',
	`name` TEXT NULL COLLATE 'utf8_unicode_ci',
	`answer` LONGTEXT NULL COLLATE 'utf8_unicode_ci',
	`is_faq` TINYINT(1) NOT NULL DEFAULT '0',
	`users_id` INT(11) NOT NULL DEFAULT '0',
	`view` INT(11) NOT NULL DEFAULT '0',
	`date` DATETIME NULL DEFAULT NULL,
	`date_mod` DATETIME NULL DEFAULT NULL,
	`begin_date` DATETIME NULL DEFAULT NULL,
	`end_date` DATETIME NULL DEFAULT NULL,
	PRIMARY KEY (`id`),
	INDEX `users_id` (`users_id`),
	INDEX `knowbaseitemcategories_id` (`knowbaseitemcategories_id`),
	INDEX `is_faq` (`is_faq`),
	INDEX `date_mod` (`date_mod`),
	FULLTEXT INDEX `fulltext` (`name`, `answer`)
)
```

Then I think this is not necessary. I testes a case when a knowbaseitem does not belongs to a category, and it worked